### PR TITLE
194 add support for instant updates on message receipts

### PR DESCRIPTION
**Pull request by nanu-c**
This adds the missing bits that makes it smooth

### CODE REVIEW

File: axolotl-web/src/store/store.js [hunk 1]
```diff
@@ -246,6 +265,9 @@ export default new Vuex.Store({
         else if (Object.keys(messageData)[0] == "MessageRecieved") {
           this.commit("SET_MESSAGE_RECIEVED", messageData["MessageRecieved"]);
```
Comment (Flaburgan) [inline axolotl-web/src/store/store.js:265]: I had a `Uncaught TypeError: can't access property "push", tmpList is null    SET_MESSAGE_RECIEVED store.js:167` here in my conversation with all messages failing. Not sure how to reproduce exactly.

**Comment by Flaburgan** [comment]
It looks like it works well for the sending / sent / received states, but not sending -> error, which always stays on sending until I refresh the page.

File: axolotl-web/src/store/store.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ Vue.use(Vuex)
 export default new Vuex.Store({
   state: {
     chatList: [],
-    messageList: [],
+    messageList: {},
```
Comment (olof-nord) [inline axolotl-web/src/store/store.js:9]: What about renaming `messageList` to `messages`?
Is the list part referring to the data type, or that the content is not a single item?

Follow-up question:
If the messageList object only contains the child Messages, could the structure be flattened?
`state.messageList.Messages`-> `state.Messages`

File: axolotl-web/src/store/store.js [hunk 3]
```diff
@@ -174,6 +177,22 @@ export default new Vuex.Store({
       })
 
     },
+    SET_MESSAGE_UPDATE(state, message) {
+      if (state.messageList.ID == message.ChatID) {
+        var index = state.messageList.Messages.findIndex(m => {
+          if (m.ID === message.ID) return true
```
Comment (olof-nord) [inline axolotl-web/src/store/store.js]: Could the comparison be shortened, and returned straight away?

`return m.ID === message.ID`

File: axolotl-web/src/store/store.js (hunk 2 above)
Comment (nanu-c) [reply axolotl-web/src/store/store.js:9]: The data structure contains everything the MessageList page needs to render, so it's like that
image

File: axolotl-web/src/store/store.js (hunk 3 above)
Comment (nanu-c) [reply axolotl-web/src/store/store.js]: I could be, but that's a coding style i am not used to. We can change that.

File: axolotl-web/src/store/store.js (hunk 1 above)
Comment (nanu-c) [reply axolotl-web/src/store/store.js:265]: mhm, that's strange, i need to try to reproduce.

File: axolotl-web/src/store/store.js (hunk 3 above)
Comment (olof-nord) [reply axolotl-web/src/store/store.js]: I dont want to push my opinions, it was more an attempt to introduce less code. Possibly a few lines in a CONTRIBUTING.md could make sure we are all aligned?

File: axolotl-web/src/store/store.js (hunk 2 above)
Comment (olof-nord) [reply axolotl-web/src/store/store.js:9]: Fantastic explanation, really like the screenshot.
What about leaving the `List` suffix for something which sounds less like a type, like `Items`?

Seeing the object messageList leads easily to make the assumption that it is a list, but in fact its an object containing a list.

File: app/sender/messageSender.go [hunk 4]
```diff
@@ -37,14 +37,23 @@ func SendMessageHelper(to, message, file string) (error, *store.Message) {
 		m.ExpireTimer = session.ExpireTimer
 		_, savedM := store.SaveMessage(m)
 
-		go SendMessage(session, m)
+		go func() {
+			mNew, err := SendMessage(session, m)
+			if err != nil {
+				// log.Errorln("[axolotl] Error sending message:" + err.Error())
```
Comment (Flaburgan) [inline app/sender/messageSender.go]: This should be uncommented I guess?

File: axolotl-web/src/store/store.js (hunk 3 above)
Comment (Flaburgan) [reply axolotl-web/src/store/store.js]: Let's do that change here indeed.

File: axolotl-web/src/store/store.js (hunk 1 above)
Comment (Flaburgan) [reply axolotl-web/src/store/store.js:265]: I didn't have it again at the moment, I'll open another issue if it happens again.

File: axolotl-web/src/store/store.js (hunk 2 above)
Comment (Flaburgan) [reply axolotl-web/src/store/store.js:9]: That's a good suggestion, but I'd like the refactor to be done in another PR, the main one about message status is becoming big enough.

**Comment by Flaburgan** [comment]
@nanu-c  Everything now works correctly, I'll let you solve the two remaining remarks and then I'll merge this.

File: app/sender/messageSender.go (hunk 4 above)
Comment (nanu-c) [reply app/sender/messageSender.go]: we have already a error message down the stack, we don't need to log it twice, so i removed it.

**Pull request merged**